src/onnx/parse_split.cpp

```diff
@@ -49,75 +49,139 @@ struct parse_split : op_parser<parse_split>
+            }
+            // last slice: slice(input, starts = {n * chunk_size}); ends = max_int, axes =
+            // {tuned_axis}
+            ret_ins.at(num_outputs - 1) = info.add_instruction(
+                make_op("slice",
+                        {{"axes", {tuned_axis}}, {"ends", {std::numeric_limits<int64_t>::max()}}}),
+                args[0],
+                info.add_instruction(
+                    make_op("mul"),
+                    chunk_size,
+                    info.add_literal(literal{int64_scalar_shape, {num_outputs - 1}})));
+            return ret_ins;
         }
-        // no split attribute, input is equally divided
         else
         {
-            std::size_t num_outputs = info.num_outputs;
-            // the num_outputs attribute seems to be redundant since we already have
-            // node_info::num_outputs, but we can still perform an error check
-            if(contains(info.attributes, "num_outputs"))
+            // either static shape or fixed dynamic_dimension for split axis
+            auto tuned_axis_len = input_shape.to_static(0).lens().at(tuned_axis);
+            std::vector<int64_t> vec_splits;
+            if(contains(info.attributes, "split"))
+            {
+                literal s = parser.parse_value(info.attributes.at("split"));
+                s.visit([&](auto v) { vec_splits.assign(v.begin(), v.end()); });
+            }
+            else if(args.size() == 2)
             {
-                num_outputs =
-                    parser.parse_value(info.attributes.at("num_outputs")).at<std::size_t>();
-                if(num_outputs != info.num_outputs)
+                auto s = args[1]->eval();
+                check_arg_empty(s, "PARSE_SPLIT: non-constant `split` input is not supported");
+                s.visit([&](auto v) { vec_splits.assign(v.begin(), v.end()); });
+            }
+            // no split attribute, input is equally divided
+            else
+            {
+                std::size_t num_outputs = info.num_outputs;
+                // the num_outputs attribute seems to be redundant since we already have
+                // node_info::num_outputs, but we can still perform an error check
+                if(contains(info.attributes, "num_outputs"))
+                {
+                    num_outputs =
+                        parser.parse_value(info.attributes.at("num_outputs")).at<std::size_t>();
+                    if(num_outputs != info.num_outputs)
+                    {
+                        MIGRAPHX_THROW("PARSE_SPLIT: num_outputs attribute " +
+                                       std::to_string(num_outputs) +
+                                       " doesn't match actual number of outputs " +
+                                       std::to_string(info.num_outputs) + "!");
+                    }
+                }
+                if(tuned_axis_len % num_outputs == 0)
```

Comment:
Style: the if/else clause can be compressed by just having a brief if-clause for the last bucket.
```
vec_splits(num_outputs, chunk_size);
auto last = tuned_axis_len%num_outputs;
if(last) vec_splits[num_outputs-1] = last;
```

Reply:
Doesn't work, logic happens such that the last bucket is the smallest and can be 0.